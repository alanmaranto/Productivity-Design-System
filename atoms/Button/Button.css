:root {
  --medium-purple: #9879e9;
  --white: #fff;
  --white-lilac: #e8e8e8;
}

.button {
  border: 1px solid transparent;
  padding: 10px 85px;
  border-radius: 30px;
  box-shadow: 0px 4px 5px var(--white-lilac);
}

.type-primary {
  background: var(--medium-purple);
  color: var(--white);
}

.type-secondary {
  background: transparent;
  color: var(--medium-purple);
  border: 1px solid var(--medium-purple);
}
