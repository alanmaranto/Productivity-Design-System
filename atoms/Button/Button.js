import React from "react";
import PropTypes from "prop-types";
import { options } from "./constants";
import Heading from "../Heading";

import styles from "./Button.module.css";
import withStyles from "../../hoc/withStyles";

const Button = ({
  children,
  type,
  getStyles,
  isInline,
  onClick,
  isMuted,
  addons,
}) => (
  <button
    className={getStyles("button", ["type"], {
      "is-inline": isInline || type === "tertiary",
      "is-muted": isMuted && type === "primary",
    })}
    onClick={onClick}
  >
    {addons && addons.prepend}
    <Heading color={type === "primary" ? "inverted" : "primary"}>
      {children}
    </Heading>
    {addons && addons.prepend}
  </button>
);

Button.propTypes = {
  children: PropTypes.string.isRequired,
  type: PropTypes.oneOf(options.types),
  getStyles: PropTypes.func.isRequired,
  isInline: PropTypes.bool,
  onClick: PropTypes.func,
  isMuted: PropTypes.bool,
  addons: PropTypes.shape({
    prepend: PropTypes.node,
    append: PropTypes.node,
  }),
};

Button.defaultProps = {
  type: "primary",
  isInline: true,
  onChange: () => {},
  getStyles: () => {},
};

export default withStyles(styles)(Button);
