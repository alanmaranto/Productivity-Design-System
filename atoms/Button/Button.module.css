.button {
  display: inline-flex;
  width: 100%;
  min-width: max-content;
  height: 45px;
  align-items: center;
  justify-content: center;
  padding: 10px 20px;
  border: var(--border-width-default) solid var(--color-transparent);
  border-radius: var(--border-radius-lg);
  box-shadow: 0px 4px 5px var(--box-shadow-sm);
  cursor: pointer;
  font-family: var(--font-family-sans);
  font-size: var(--font-size-md);
  font-weight: var(--font-weight-bold);
  transition: all 0.5s ease;
  /* max-width: var(--button-max-width); */
  /* white-space: nowrap; */ /* Bloquear palabras en dos lineas en botones */
}

.button:focus {
  box-shadow: 0 0 0 1px var(--color-primary), 0 0 10px 0 var(--color-primary);
  outline: none;
}

.button:hover {
  filter: brightness(1.1);
}

.button:active {
  filter: brightness(0.9);
}

.type-primary {
  background: var(--color-primary);
  /* color: var(--color-white); */
  color: var(--color-primary-inverted);
}

.type-primary:focus {
  box-shadow: 0 0 0 1px var(--color-primary-inverted),
    0 0 10px 0 var(--color-brand-medium-purple);
}

.type-secondary {
  border: var(--border-width-default) solid var(--color-primary);
  background: var(--color-transparent);
  color: var(--color-primary);
}

.type-tertiary {
  height: 25px;
  padding: 0;
  border-bottom: 1px solid var(--color-primary);
  /* --color-transparent: blue; */
  background: var(--color-base-transparent);
  border-radius: var(--border-radius-none);
  box-shadow: var(--shadows-none);
  color: var(--color-primary);
}

.type-tertiary:focus {
  box-shadow: 0 10px 10px -8px var(--color-primary);
}

.button.is-inline {
  max-width: max-content;
}

.is-muted {
  background: var(--color-primary-muted);
}
