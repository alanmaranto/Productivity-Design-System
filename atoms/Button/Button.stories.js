import Button, { options, styles } from '.';
import {
  getOptionsArgTypes,
  getTemplate,
  getListTemplate,
} from '../../helpers/storybook';

export default {
  title: 'Atoms/Button',
  component: Button,
  args: {
    children: 'You shall not pass!',
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
    children: { control: 'text' },
    isMuted: { description: 'It requires `type`to be "primary"' },
    onClick: { defaultValue: null },
  },
};

const Template = getTemplate(Button, styles);
const ListTemplate = getListTemplate(Button, styles);

export const Default = Template.bind({});
export const Types = ListTemplate.bind({});
Types.args = {
  items: options.types.map((type) => ({ type })),
};

export const Inline = Template.bind({});
Inline.args = {
  isInline: true,
};

export const Muted = Template.bind({});
Muted.args = {
  type: 'primary',
  isMuted: true,
};
