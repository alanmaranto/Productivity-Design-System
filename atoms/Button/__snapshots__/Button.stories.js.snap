// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Button Addon Append 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button"
    onClick={[Function]}
  >
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
    But ill-tempered Thengel.
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Addon Prepend 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button"
    onClick={[Function]}
  >
    Dare Erebor forgive most.
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Default 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button"
    onClick={[Function]}
  >
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Inline 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button is-inline"
    onClick={[Function]}
  >
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Muted 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary is-muted"
    onClick={[Function]}
  >
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
</div>
`;

exports[`Storyshots Atoms/Button Types 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <button
    className="button type-primary"
    onClick={[Function]}
  >
    <header
      className="color-inverted size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
  <button
    className="button type-secondary"
    onClick={[Function]}
  >
    <header
      className="color-primary size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
  <button
    className="button type-tertiary is-inline"
    onClick={[Function]}
  >
    <header
      className="color-primary size-md weight-bold heading"
    >
      You shall not pass default!
    </header>
  </button>
</div>
`;
